File: app/src/main/java/com/bove/martin/pexel/adapter/SearchAdapter.java
package com.bove.martin.pexel.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bove.martin.pexel.R;
import com.bove.martin.pexel.model.Foto;
import com.bove.martin.pexel.model.Search;
import com.bumptech.glide.Glide;

import java.util.List;

/**
 * Created by Martín Bove on 31/05/2018.
 * E-mail: mbove77@gmail.com
 */
public class SearchAdapter extends RecyclerView.Adapter<SearchAdapter.ViewHolder> {
    private List<Search> searches;
    private int layout;
    private OnItemClickListener listener;

    public SearchAdapter(List<Search> searches, int layout, OnItemClickListener listener) {
        this.searches = searches;
        this.layout = layout;
        this.listener = listener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        //Inflamos la vista
        View view = LayoutInflater.from(parent.getContext()).inflate(layout,parent,false);
        ViewHolder viewHolder = new ViewHolder(view);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.bind(searches.get(position), listener);
    }

    @Override
    public int getItemCount() {
        return searches.size();
    }

    // ViewHolder
    public class ViewHolder extends RecyclerView.ViewHolder {
        private ImageView imageViewFoto;
        private TextView searchTerm;

        public ViewHolder(View itemView) {
            super(itemView);
            imageViewFoto = itemView.findViewById(R.id.imageSearch);
            searchTerm = itemView.findViewById(R.id.searchTerm);
        }

        // Aca esta la logica de remplazo de datos y asiganacion de eventos
        public void bind(final Search search, final OnItemClickListener listener) {

            Glide.with(itemView)
                    .load(search.getPhoto())
                    .placeholder(R.drawable.placeholder)
                    .into(imageViewFoto);

            searchTerm.setText(search.getSearchInSpanish());

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    listener.onItemClick(search, getAdapterPosition());
                }
            });
        }
    }

    // Listener para comunicar el item click
    public interface OnItemClickListener {
        void onItemClick(Search search, int posicion);
    }
}

File: app/src/main/java/com/bove/martin/pexel/presentation/adapters/SearchAdapter.kt
package com.bove.martin.pexel.presentation.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bove.martin.pexel.R
import com.bove.martin.pexel.domain.model.Search
import com.bumptech.glide.Glide

/**
 * Created by Martín Bove on 31/05/2018.
 * E-mail: mbove77@gmail.com
 */
class SearchAdapter(private val searches: List<Search>, private val layout: Int, private val listener: OnSearchItemClickListener) : RecyclerView.Adapter<SearchAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        //Inflamos la vista
        val view = LayoutInflater.from(parent.context).inflate(layout, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(searches[position], listener)
    }

    override fun getItemCount(): Int {
        return searches.size
    }

    // ViewHolder
    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val imageViewFoto: ImageView
        private val searchTerm: TextView

        // Aca esta la logica de remplazo de datos y asiganacion de eventos
        fun bind(search: Search, listener: OnSearchItemClickListener) {

            // if photo is null is the curated especial item
            Glide.with(itemView)
                    .load(search.photo)
                    .placeholder(R.drawable.placeholder)
                    .into(imageViewFoto)
            searchTerm.text = search.searchInSpanish
            itemView.setOnClickListener { listener.onSearchSuggestItemClick(search, bindingAdapterPosition) }
        }

        init {
            imageViewFoto = itemView.findViewById(R.id.imageSearch)
            searchTerm = itemView.findViewById(R.id.searchTerm)
        }
    }

    // Listener para comunicar el item click
    interface OnSearchItemClickListener {
        fun onSearchSuggestItemClick(search: Search, posicion: Int)
    }
}